app/controllers/institutions_controller.rb

```diff
@@ -128,15 +128,15 @@ def new_from_invite_data
     end
   end
 
+  private
+
   def accept_pending_invite
-    unless @pending_invite.nil? and current_user.email != @pending_invite.invited_user_email
+    if @pending_invite && current_user.email == @pending_invite.invited_user_email
```

Comment:
This was an actual logic error, probably caused by confusion about double negation (it should've been `or`, not `and`). I applied a De Morgan transform to make the expression easier to understand.